eslint-plugin-expensify/no-negated-variables.js

```diff
@@ -7,7 +7,20 @@ const message = require('./CONST').MESSAGE.NO_NEGATED_VARIABLES;
  * @returns {Boolean}
  */
 function isFalsePositive(string) {
-    return _.some(['notification', 'notch'], falsePositive => string.toLowerCase().includes(falsePositive));
+    const matches = /(.*?)(?:[nN](?:otification|otch|ote))+(.*)/gm.exec(string);
```

Comment:
Explanation:

1. `(.*?)` – This uses "lazy" matching to capture everything up until the first instance of the next part of the pattern.
1. `[nN](?:otification|otch|ote)` – This matches `notification`, `Notification`, `note`, `Note`, `notch`, `Notch`
    1. We then wrap this in another non-capturing group and match one or more: `(?:<expression>)+`. This matches `notificationNote` or `noteNotification`
1. Lastly, we capture everything after in that in the string with `(.*)` ​


Then after that, we captured everything before and after the matched word or words – we make sure that prefix or suffix does not include `Not` or `not`.

Reply:
Ok, I'm not sure if I get it, but that's ok.

Curious if we can maybe abstract this a bit to make it easier to modify in the future for non-regex heads like me?

If we wanted to make sure something like `notableNotions` didn't trigger a false positive would I just do something like:

```js
    const matches = /(.*?)(?:[nN](?:otification|otch|ote|otion|otable))+(.*)/gm.exec(string);
```

Or is that wrong?

Reply:
Yeah, I can abstract this a bit to make it easier to add more buzz-words 🙂

Reply:
Awesome! Thanks, have a feeling we will be adding some more things here in the future :)

Reply:
Updated